src/app/user/components/user-info/user-info.component.ts

```diff
@@ -0,0 +1,103 @@
+import {Component, OnInit} from '@angular/core';
+import {AbstractControl, FormControl, FormGroup, Validators} from '@angular/forms';
+import {Store} from '@ngrx/store';
+import {getUser, UserState} from '../../reducers/user.reducer';
+import {User} from '../../../../entities/user.model';
+import {update, updateWithoutPhone} from '../../actions/user.actions';
+import firebase from 'firebase';
+import {AuthenticateService} from '../../services/authentication.service';
+import {WindowService} from '../../../services/auth/window.service';
+import auth = firebase.auth;
+
+
+@Component({
+  selector: 'app-user-info',
+  templateUrl: './user-info.component.html',
+  styleUrls: ['./user-info.component.scss']
+})
+export class UserInfoComponent implements OnInit {
+  editDetailsForm: FormGroup;
+  birthday: Date;
+  user: User = null;
+  windowRef: any;
+
+  constructor(private store: Store<UserState>,
+              private authenticateService: AuthenticateService,
+              private windowService: WindowService) {
+    this.windowRef = windowService.windowRef;
+  }
+
+  ngOnInit(): void {
+    this.editDetailsForm = new FormGroup({
+      firstNameForm: new FormControl(null, [Validators.required, this.noWhitespaceValidator]),
+      lastNameForm: new FormControl(null, [Validators.required, this.noWhitespaceValidator]),
+      emailForm: new FormControl(null, [Validators.required, Validators.email]),
+      phoneNumberForm: new FormControl(null, [Validators.required, Validators.minLength(10)])
+    });
+
+    this.initializeUserDetails();
+    this.initializeRecaptcha();
+  }
+
+
+  initializeUserDetails(): void {
+    this.store.select(getUser).subscribe(user => {
```

Comment:
This is a memory leak. When you use subscribe you always need to unsubscribe.
Create an `onDestroy$' subject and use `takeUntil(onDestroy$)` 

check out this article:  - 3. Let’s Get Declarative With takeUntil
https://medium.com/angular-in-depth/the-best-way-to-unsubscribe-rxjs-observable-in-the-angular-applications-d8f9aa42f6a0

Reply:
done.